### PR TITLE
[c++/python/r] Move group-create to `libtiledbsoma`

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:**

#1182

**Changes:**

Replace `tiledb_group_create` (from TileDB-Py and TileDB-R) with `libtliedbsoma::SOMAGroupWriter::create`.

**Notes for Reviewer:**

The group-create is a small, isolable change that doesn't touch the layered complexities of `_tdb_handles.py`. This is a good place to set a pattern for the second `libtiledbsoma` class after the venerable `SOMAArrayReader`. More simply put, this is a quick and easy checkbox we can check off on #922.

Note that `SOMAGroupReader` is underway on #1122.

### CODE REVIEW

File: apis/python/tests/test_dataframe_index_columns.py [hunk 1]
```diff
@@ -1812,70 +1812,100 @@ def test_types_write_errors(
             ["int32"],
             None,
             [pa.array([3202, 3203])],
-            RuntimeError,  # Static type (INT64) does not match expected type (INT32)
+            (
+                tiledb.cc.TileDBError,
```
Comment (johnkerl) [inline apis/python/tests/test_dataframe_index_columns.py:1816]: not caused by this PR, but necessary today in my sandbox (either on `main` or this branch)

**Review by eddelbuettel — COMMENTED** [review]
(Concerning the R and C++ parts)  Seems fine but I cannot help but wonder if it is a longer-time winning strategy to duplicate what seems like identical code in two place.  But that decision was made above my paygrade so ...

Will hold on approving until I seen green everywhere.

**Comment by johnkerl** [comment]
Thanks for the update @Shelnutt2 ! Moving to draft.

**Comment by johnkerl** [comment]
> (Concerning the R and C++ parts) Seems fine but I cannot help but wonder if it is a longer-time winning strategy to duplicate what seems like identical code in two places. But that decision was made above my paygrade so ...
> 
> Will hold on approving until I seen green everywhere.

@eddelbuettel I put the `SOMAGroupWriter` logic in a common place, then did the Python bindings from the pybind11 file and the R bindings from the Rcpp file -- I think these are necessary ...

**Comment by johnkerl** [comment]
Closing on discussion with @nguyenv

**Pull request closed without merging**